src/kmer.py

```diff
@@ -7,25 +7,41 @@ def kmer(x: str, k: int) -> list[str]:
 
     >>> kmer('agtagtcg', 3)
     ['agt', 'gta', 'tag', 'agt', 'gtc', 'tcg']
-
-    FIXME: do you want more tests here?
     """
-    ...
+    kmers=[]
+    for i in range(0, len(x)-k+1):
+        kmerr=x[i:i+k]
+        kmers.append(kmerr)
+    return kmers
+
 
 
 def unique_kmers(x: str, k: int) -> list[str]:
     """
-    Computer all unique k-mers of x.
-
-    FIXME: do you want more tests here?
+    >>> unique_kmers('abcd', 2)
+    ['ab', 'bc', 'cd']
     """
-    ...
+    kmers=[]
+    for i in range(0, len(x)-k+1):
+        kmerr=x[i:i+k]
+        if kmerr not in kmers:
```

Comment:
This test is a linear search. It makes the function run in $O(n^2)$. If you used a `set` for `kmers` instead, you could test in $O(1)$ and get an $O(n)$ function. Then you could translate the set into a list again just before you return

```python
kmers = set()
...
return list(kmers)
```